Replace lunardate with lunarcalendar

derlikh-smart opened the pull request:
refs #536 

- [ ] Tests with a significant number of years to be tested for your calendar.
- [ ] Changelog amended with a mention describing your changes. **Note** *Please do NOT change the version number here. It's the project maintainers' duty.*

brunobord reviewed (changes requested):
you've added an unused import (`DateNotExist` exception). Either you'll have to use it somewhere in the code, or get rid of it.

derlikh-smart commented:
> you've added an unused import (`DateNotExist` exception). Either you'll have to use it somewhere in the code, or get rid of it.

Good catch!
I removed the unused import.

derlikh-smart commented:
@brunobord can you approve the workflow to check if everything is OK now?

OriginalPenguin commented:
This is an important pull request.  Workalendar as an MIT licensed project really shouldn't be pulling in any GPL code, since doing so is a violation of the GPL.

JensAstrup commented:
What is the status of this PR, as mentioned in the comment above this is a rather important change given the licensing

brunobord commented:
could you please rebase with the latest master? I've removed the (now obsolete) Python 3.6 test suite, and added Python 3.10 & 3.11 compatibility.

brunobord commented:
> This is an important pull request. Workalendar as an MIT licensed project really shouldn't be pulling in any GPL code, since doing so is a violation of the GPL.

This contribution will be integrated when it's ready.

But again, I may not be a lawyer, but I do think that *using* GPL code by downloading a third-party library and making calls to it using its API has nothing to do with a violation of the GPL. We don't **embed** GPL code in workalendar. We didn't **add changes** to GPL code.

It's as if you were forcing people to release EVERY BIT OF THEIR WORK using the terms of the GPL, because they're running any GNU/Linux distro ; because the kernel is GPL, and thus you PC is making calls to the kernel via any program (your text editor, your network manager, you shell), and by "pulling in GPL code", you'd have to publish the source code of every tiny thing your PC does.

The MIT and the GPL licences are, as far as I can say, compatible with each other.

There's IMO, a difference between using the source code + making changes to it or making static links between binaries *vs.* making a dynamic link using the principle of imports as we do in Python.

derlikh-smart commented:
> could you please rebase with the latest master? I've removed the (now obsolete) Python 3.6 test suite, and added Python 3.10 & 3.11 compatibility.

Rebase is done.

orsinium commented:
@brunobord [I'm also not a lawier](https://en.wikipedia.org/wiki/IANAL) and my message cannot be considered legal advice. Still, digging into the mess of software licensing is my hobby, and I understand this mess to some extent. I've been watching the story, and this is my take on it.

You're 100% right. How we use dependencies in Python is called "dynamic linking". You don't distribute a binary of the GPL-licensed library, you don't modify its source code, and you don't vendorize it. Hence you can legally use it without any effect on licensing of your own project. This is known as [GPL linking exception](https://en.wikipedia.org/wiki/GPL_linking_exception). Nobody can legally force you to change the project license or remove the GPL dependency from the project.

**However** there is more to it. MIT allows private usage without disclosing the project source code. GPL doesn't. If I use workalendar in a proprietary commercial product, I am forced also to install the GPL-licensed library for workalendar to work. And that's where the issue lies. If I provide some kind of service to users, either as a web service or a standalone application, I can't use in it workalendar without also using the GPL-licensed library and hence violating that library's license.

I'm joining the discussion because I have on my hands a big proprietary project that internally uses a fork of workalendar with dependency on lunardate removed. However, maintaining the fork is a hassle, and it already drags behind quite a lot. **Please, let's get this PR merged**. Either replacing the dependency like in this PR or making it optional as suggested by the initial issue is required for using workalendar in a proprietary project.

CreativePhilip commented:
Hello @brunobord I also have some substantial interest in getting this PR merged&released, however this issue looks like it's a bit stale. Is there any way that I can assist to perhaps move this forward so we can get this merged ?
